### PR TITLE
New multilanguage system

### LINKED ISSUE #244: Change localization process
This is something we briefly discussed in the past days. The way the website is translated now is a beginning, but not optimal. Some translations are in html files, other in yml and some sections need to be updated for each language. That makes working on translations hard and unmaintainable in the mid-long term.

As i suggested in past i think we should implement a better workflow. I think using the [jekyll-multiple-languages-plugin](https://github.com/kurtsson/jekyll-multiple-languages-plugin) to manage the localization of the webste is the way to go.

SInce this approach received some positive feedback in the past I already started to work on this. It will take some time because requires a complete change of the structure of the repo and of most of the files, but after, the maintainance of the repo will be easier and the structure will be consistent for all languages.

Feel free to assign this issue to me

### PR DESCRIPTION
**Pull request by erciccione**
## New multilanguage system

This PR adds complete the refactor of the website started with #256 and adds the new multilanguage system which is based on a plugin: [jekyll-multiple-languages-plugin](https://github.com/kurtsson/jekyll-multiple-languages-plugin). With this new system, it will be easier to add and manage new languages/locales since it's not required to duplicate all files necessaries, but only about 3, which will not be a new structure, as it was before, but will work on top of the existent one.
Closes #244 

### New structure
Now all languages are inside `_i18n`. Each language will have its own `LANG.yml` file and a `LANG` folder. which can be used for language/locale specific resources.

The base structure of the main pages is now composed by variables. This allowed me to remove 56 duplicated pages, now we need only the 8 in the root folder (`community.md`, `dao.html`, 'downloads.html', `faq.html`, `index.html`, `markets.html`, `stats.html`, `vision.md`). Most of them were used only for their front matter, now they are the base for all languages.

Two different structures were used: one for the English side (e.g `/_data/dao_content.yml`) and one for the other languages (e.g. `_data/dao_content_tr.yml`). Now the structure is the same for all languages, which means all `_tr.yml` files were removed.

If a string is edited in the `en.yml` template (which is used to check against all other languages), the correspondent variable must be resetted for all languages, because jekyll only cares that the translated variable exists, it cannot know if it's updated or not.

### Adding a new language
See changes in readme: 973bf7a 

### To note
- While building the website you will see some warnings saying
```
Missing i18n key: fr:1.1.7
Using translation '' from default language: en
```
 This is a result of using the variable `client_version` in `_config.yml`, which is not picked up by the other languages, so jekyll complains. It doesn't make any sense to move it in `en.yml` because it has to remain the same for all languages, so we can keep it as it is for now.

- If a variable is added to the english file, but not on the other languages, jekyll will automatically use the english version and will complain about it while building.

### CODE REVIEW

**Comment by erciccione** [comment]
PR ready.

**Comment by wiz** [comment]
This PR fails to deploy https://app.netlify.com/sites/bisq-website/deploys/5d8f2d86a94a7a000a3cd8c4

**Comment by erciccione** [comment]
@wiz This PR updates bundler from 2.0.1 to 2.0.2. The deploy settings need to be updated.

```
Activating bundler (2.0.2) failed:
11:53:38 AM: Could not find 'bundler' (= 2.0.2) - did find: [bundler-2.0.1,bundler-1.17.3]
11:53:38 AM: Checked in 'GEM_PATH=/opt/buildhome/.rvm/gems/ruby-2.3.6:/opt/buildhome/.rvm/gems/ruby-2.3.6@global', execute `gem env` for more information
11:53:38 AM: To install the version of bundler this project requires, run `gem install bundler -v '2.0.2'`
```

**Comment by wiz** [comment]
Did you test this on netlify?

**Comment by erciccione** [comment]
No, i tested it locally. The errors i posted come from the log you linked to.

**Comment by wiz** [comment]
Okay so can you signup on netlify using "Login with GitHub", connect your work repo/branch and when you figure out how to get it deploying, either push another commit to this PR branch or tell us how what netlify settings to change?

**Comment by erciccione** [comment]
I would prefer to not register to netlify if not necessary. Seems like it's asking a lot of info and authorizations that i prefer to not give, since it's a tool i don't use.

I think the problem can be solved by installing an updated version of bundler (2.0.2) on the netlify build machine. Anyway, the upgrade of bundler should not be needed for the new plugin to work, i'm gonna downgrade it back to 2.0.1 and force push. That should solve the issue.

**Review by wiz — COMMENTED** [review]
The code changes look okay, just need to fix the deploy errors and then we can starting testing on a staging netlify instance. @m52go might want to double-check the things removed from _layouts/default.html to make sure that doesn't affect the layout of the blog.

**Comment by wiz** [comment]
Netlify is a type of PaaS cloud that uses a pre-built OS images, so yeah just downgrade bundler back to 2.0.1 if possible. Please don't force push since I already reviewed your PR, it just needs that one-liner fix right?

**Comment by erciccione** [comment]
>  Please don't force push since I already reviewed your PR, it just needs that one-liner fix right?

Sorry i already did, but i don't see any other solution since the commit downgrading bundler would have to be squashed anyway. GitHub shows the diff between versions when squashing, so that shouldn't be an issue.

I guess the netflify build needs to be triggered again.

**Comment by wiz** [comment]
Okay netlify staging URL is working now at https://deploy-preview-260--bisq-website.netlify.com/ but seems there are some errors visible right away so please test on netlify and let us know when ready for re-review

Screen Shot 2019-09-29 at 18 10 46

**Comment by erciccione** [comment]
@wiz Could you specify how to reproduce that error? i don't see it in any of the languages in the netlify live version you linked to or in my local build.

screen_bisq-website

**Comment by wiz** [comment]
I'm testing using Chrome on macOS and it appears without doing anything.

But just browsing around the staging site I see there are quite a few errors where wrong images are used, wrong strings are used, strings are missing, etc. - please go through every page (using each locale) on the staging URL and compare it against the production URL and make things match as much as possible.

**Comment by wiz** [comment]
Left is production, Right is staging:
Screen Shot 2019-09-29 at 18 24 59
Screen Shot 2019-09-29 at 18 25 11
Screen Shot 2019-09-29 at 18 25 47
Screen Shot 2019-09-29 at 18 26 22

**Comment by erciccione** [comment]
I tested throughly this PR on Chromium and Firefox, which both show everything correctly. Jekyll doesn't show any warning or error even in verbose mode. I don't see any of the issues you are experiencing, except for the "Vision" title (that's my bad, i left it in English on purpose but i now see other languages translate it) and the missing localized picture in your last screenshot. I'm starting to think there are some hardcoded settings in the netlify machine which conflicts with the plugin, since there cannot be missed strings or the multilingual plugin would complain.

This PR changes the structure of the website, i suggest to test it locally, to make sure the configuration of the third part integration (netlify) is not messing up something.

I guess i will have to make an account on netlify anyway and see what is wrong with the setup.

edit: Now i understood what you mean with strings not appearing. Some errors are correct, but other i still seem unable to reproduce them. Will push some updates soon and then check out netlify.

**Comment by erciccione** [comment]
I fixed everything (i forgot to update the footer page and i missed a block in the DAO page) except the untranslated picture and the messed up format in the DAO page. I'm investigating, will update this PR when solved.

**Comment by erciccione** [comment]
Fixed everything (i misinterpreted the plugin, i thought it was handling localized images, but it wasn't ) except:
- The untranslated `[Exchanges]` in the Japanese *Vision* page is not an error. I guess who translated that page forgot it.
- The mismatching layout of the DAO page. I still cannot reproduce it locally. I'l keep investigating.

**Comment by wiz** [comment]
Screen Shot 2019-09-29 at 22 11 05

**Comment by wiz** [comment]
locale selector URLs are missing relative links, and missing hreflang attribute
Screen Shot 2019-09-29 at 22 12 45

File: _includes/main_nav.html [hunk 1]
```diff
@@ -29,7 +34,7 @@
 
       <ul class="navbar-nav ml-auto">
 
-          {% for item in site.data.main_nav.links %}
+          {% for item in site.data.main_nav[site.lang].links %}
```
Comment (m52go) [inline _includes/main_nav.html:33]: Current page links in the nav aren't underlined green for non-English pages.

**Comment by wiz** [comment]
Screen Shot 2019-09-29 at 22 15 44

**Comment by m52go** [comment]
Sort locale selector by name of language.
Screenshot from 2019-09-29 09-19-46

**Comment by m52go** [comment]
Capitalize "Private" on front page (this was an existing flaw).
Screenshot from 2019-09-29 09-17-58

**Comment by m52go** [comment]
Version missing on front page.
Screenshot from 2019-09-29 09-22-33

**Comment by wiz** [comment]
Screen Shot 2019-09-29 at 22 22 44

**Comment by m52go** [comment]
DAO box alignment.
Screenshot from 2019-09-29 09-24-56

**Comment by wiz** [comment]
Footer link for "Downloads" has 2 slashes `<a href="//downloads">` should be 1 slash

**Comment by wiz** [comment]
```
@@ -638,24 +681,26 @@
 
   <h2>How to Get Started</h2>
 
-  <div class="row align-items-start">
+  <div class="row align-t homepage_contents-start">
     <div class="col-12 col-md-4">
 
       <div class="grey">
         With Bisq, you can start trading in less than 10 minutes:
       </div>
       <ol>
```

**Comment by m52go** [comment]
Japanese translations missing toward bottom of DAO page.

Screenshot from 2019-09-29 09-35-17
Screenshot from 2019-09-29 09-35-25

**Comment by wiz** [comment]
```
@@ -379,7 +423,7 @@
                                       <a class="dl-win64" href="https://bisq.network/downloads/v1.1.7/Bisq-64bit-1.1.7.exe">Download for windows</a>
                                   </div>
                                   <div class="float-right">
-                                      <a class="dl-win64" href="https://bisq.network/downloads/v1.1.7/Bisq-64bit-1.1.7.exe">.exe</a>
+                                      <a class="dl-win64" href="https://bisq.network/downloads/v1.1.7/Bisq-64bit-1.1.7.exe">
                                   </div>
                               </div>
 
```

**Comment by wiz** [comment]
```
@@ -155,24 +197,6 @@
           
 
           
-          <li class="downloads-mobile nav-item ">
-            
-              <a class="nav-link" href="/community/">Community</a>
-            
-          </li>
-          
-          <li class="downloads-mobile nav-item ">
-            
-              <a class="nav-link" href="/blog/">Blog</a>
-            
-          </li>
-          
-          <li class="downloads-mobile nav-item ">
-            
-              <a class="nav-link" rel="noopener" target="_blank" href="https://docs.bisq.network/">Documentation</a>
-            
-          </li>
-          
 
 
           <li class="downloads-mobile nav-item ">
```

**Comment by wiz** [comment]
```
@@ -12,19 +23,19 @@
 
         <meta name="author" content="Bisq">
 
-        <meta name="description" content="Bisq is an open-source desktop application that allows you to buy and sell bitcoins in exchange for national currencies, or alternative cryptocurrencies.">
+        <meta name="description" content="">
         <meta name="image" content="/images/bisq-og.jpg">
         <meta itemprop="name" content="Bisq">
-        <meta itemprop="description" content="Bisq is an open-source desktop application that allows you to buy and sell bitcoins in exchange for national currencies, or alternative cryptocurrencies.">
+        <meta itemprop="description" content="">
         <meta itemprop="image" content="/images/bisq-og.jpg">
         <meta name="og:title" content="Bisq - The decentralized Bitcoin exchange">
-        <meta name="og:description" content="Bisq is an open-source desktop application that allows you to buy and sell bitcoins in exchange for national currencies, or alternative cryptocurrencies.">
+        <meta name="og:description" content="">
         <meta name="og:image" content="/images/bisq-og.jpg">
-        <meta name="og:site_name" content="Bisq - The decentralized Bitcoin exchange">
+        <meta name="og:site_name" content="">
         <meta name="og:type" content="website">
         <meta property="twitter:title" content="Bisq - The decentralized Bitcoin exchange">
         <meta property="twitter:card" content="summary_large_image">
-        <meta property="twitter:site" content="false">
+        <meta property="twitter:site" content="">
         <meta name="twitter:creator" content="@bisq_network">
         <link rel="canonical" href="/">
         
```

**Comment by m52go** [comment]
English mobile menu is missing links from lean-more drop-down (e.g., documentation, blog, community).

Screenshot from 2019-09-29 09-39-00

**Comment by m52go** [comment]
"Learn more" link missing on front page (English and non-English).

Screenshot from 2019-09-29 09-42-01

**Comment by wiz** [comment]
@FKrauss can you verify the Matomo analytics code is working properly at https://deploy-preview-262--bisq-website.netlify.com/ ? I see some errors in the log here, not sure if it's related https://matomo.bisq.network/index.php?module=CoreAdminHome&action=trackingFailures&period=day&date=yesterday&idSite=1

**Comment by FKrauss** [comment]
I'll take a look at it today. I think you might see errors because the pings are not coming from the domain we set in matomo. Maybe we'll have to create a test env for matomo for these not to fail. I'll look at it and let you know

**Comment by erciccione** [comment]
I created a simple "fixes" commit with the changes. I will squash it after it's reviewed.

- [x] "Learn more" link missing on front page (English and non-English).
- [X] English mobile menu is missing links from lean-more drop-dow
- [x] Missing meta content
- [x] Missing version in homepage (under downloads)
- [x] Missing text in Japanes DAO page.
- [x] "Footer link for "Downloads" has 2 slashes <a href="//downloads"> should be 1 slash"
- [x] https://github.com/bisq-network/bisq-website/pull/260#issuecomment-536300201
- [x] *private* capitalized
- [x] Sort locale selector by name of language.
- [x] Mismatching layout in DAO
- [x] *Current page links in the nav aren't underlined green for non-English pages.*
- [x] Different font for Japanese language

I have to take a deeper look at the nav and figure out a way to make it underlined green for the other languages. ~~Also, the problem with the layout still occurs, i still haven't found the problem.~~ <- somehow the copy-pasted structure was mismatching.

Please refer to the source code when reviewing, not to the built source code, or it's much harder to understand what you are referring to and track it in the code.

**Comment by wiz** [comment]
@erciccione can you please merge master into this PR branch to fix the merge conflict, and re-squash your commits to remove the "fixes" commit? I think @m52go and I will take another look at this tomorrow.

**Comment by erciccione** [comment]
Rebased and fixed remaining points. This is ready to be reviewed, but i will take another look today and add fixes if necessary.

File: _config.yml [hunk 2]
```diff
@@ -1,13 +1,18 @@
 title: Bisq
 url: "https://bisq.network"
 markdown: kramdown
-permalink: /blog/:title/
+permalink: /blog:title/
```
Comment (m52go) [inline _config.yml]: Was this change a mistake?

Result:

Screenshot from 2019-10-02 16-24-12

File: _config.yml [hunk 3]
```diff
@@ -18,7 +23,7 @@ exclude:
   - README
   - Gemfile
 paginate: 10
-paginate_path: "/blog/page/:num/"
+paginate_path: "/blog/page:num/"
```
Comment (m52go) [inline _config.yml]: Same as above...there should be a slash.

File: _includes/main_nav.html [hunk 4]
```diff
@@ -3,23 +3,29 @@
...
+	   {% assign sorted-items = site.data.languages | sort: "name" %}
+       {% for item in sorted-items %}
           <div class="dropdown-item btn-icon">
             <div class="float-right">
-              <a href="{{ page.url }}" hreflang="{{ page.lang }}">{{ page.language }} {{ page.flag }}</a>
+               {% if item.tag == "en" %}
+               <a href="/">{{ item.name }} {{ item.flag }}</a>
+			   {% else %}
+			   <!-- we use the 'tag' variable in languages.yml to redirect to the correct page -->
+			   <a href="/{{ item.tag}}/" hreflang="{{ site.lang }}">{{ item.name }} {{ item.flag }}</a>
```
Comment (m52go) [inline _includes/main_nav.html]: Right now, choosing a new language loads the translation of the current page. With this PR, choosing a new language always loads the home page in the chosen language.

Not a deal-breaker, in my opinion, but worth pointing out. Maybe something to be fixed later.

File: index.html [hunk 5]
```diff
@@ -74,49 +69,47 @@ <h1 class="intro text-monospace display-1" style="margin-left: -.3rem;">{{ item.
...
-                              {% include os_selector_options_tr.html %}
+                              {% include os_selector_options.html %}
                           </div>
-<!-- Need to repeat the function because the if/else function seems to nullify it -->
-{% assign item = site.data.en.homepage_content %}
                       </div>
 
 
-                      <a href="/{{ page.lang }}/downloads" class="downloads-mobile btn btn-lg btn-outline-secondary btn-icon hero-btn col-sm-12 col-md-12 col-lg-3 px-0 shadow-btn ">
+                      <a href="/downloads" class="downloads-mobile btn btn-lg btn-outline-secondary btn-icon hero-btn col-sm-12 col-md-12 col-lg-3 px-0 shadow-btn "> <!-- TODO -->
```
Comment (m52go) [inline index.html]: Looks like the front-page download button always goes to the English version of the Downloads page when another language is set.

**Review by m52go — COMMENTED** [review]
I think we're getting there! Attached a couple of comments.

File: index.html (hunk 5 above)
Comment (erciccione) [reply index.html]: Forgotten TODO :) Fixing it and taking advantage of this to fix also the link of the button.

File: _includes/main_nav.html (hunk 4 above)
Comment (erciccione) [reply _includes/main_nav.html]: Yes. Now remember it was mentioned during the call. If it's not a big deal we can have this merged first and fix that issue after.

File: _config.yml (hunk 2 above)
Comment (erciccione) [reply _config.yml]: I didn't change them back after some testing. Fixing.

**Comment by erciccione** [comment]
Pushed fixes and rebased. I don't know if i will have time to fix https://github.com/bisq-network/bisq-website/pull/260#discussion_r330774278 today. In case, i will work on it very soon.

File: _includes/main_nav.html (hunk 4 above)
Comment (erciccione) [reply _includes/main_nav.html]: Just pushed the fix for this.

**Comment by erciccione** [comment]
Just pushed one minor fix (https://github.com/bisq-network/bisq-website/pull/260#discussion_r330774278). I won't touch this PR anymore unless other bugs are found. If necessary, i will open new PRs.

**Comment by m52go** [comment]
@erciccione stats page isn't showing stats captions on the bottom of the page for any languages:

Screenshot from 2019-10-07 15-18-34

Otherwise looks good to me. 

@wiz would you like to take another quick look? Would prefer to have another set of eyes approve before I merge since this is a relatively significant PR.

**Comment by erciccione** [comment]
I must have made a mistske while rebasing on #263. Pushing the fix now.

**Comment by m52go** [comment]
@erciccione did you revert the whole `_config.yml` file to a previous version when fixing the `paginate_path` variable? The `version` is now "v1.1.2". Not sure if that change triggered any other changes I'm not seeing...

**Comment by erciccione** [comment]
No, i fixed manually. What is the version "v1.1.2" you are referring to? i see correctly "1.1.7".

**Comment by m52go** [comment]
Ah shoot, sorry. I was looking at the wrong branch.

**Comment by m52go** [comment]
ACK

**Comment by m52go** [comment]
Will wait a day or so to merge in case of feedback from wiz or others.

File: _includes/main_nav.html [hunk 6]
```diff
@@ -3,23 +3,29 @@
     <a class="navbar-brand" href="/"><img src="/images/bisq-logo.svg" height="30"/></a>
```
Comment (wiz) [inline _includes/main_nav.html]: When viewing the /ja/dao/ page, clicking on the Bisq logo should take me to /ja/ not /

File: _includes/main_nav.html [hunk 7]
```diff
@@ -3,23 +3,29 @@
...
+	   {% assign sorted-items = site.data.languages | sort: "name" %}
+       {% for item in sorted-items %}
           <div class="dropdown-item btn-icon">
             <div class="float-right">
-              <a href="{{ page.url }}" hreflang="{{ page.lang }}">{{ page.language }} {{ page.flag }}</a>
+               {% if item.tag == "en" %}
+               <a href="/">{{ item.name }} {{ item.flag }}</a>
+			   {% else %}
+			   <!-- we use the 'tag' variable in languages.yml to redirect to the correct page -->
+			   <a href="/{{ item.tag}}{{ page.url }}" hreflang="{{ site.lang }}">{{ item.name }} {{ item.flag }}</a>
```
Comment (wiz) [inline _includes/main_nav.html]: hreflang should be `item.tag` not `site.lang`

File: _includes/main_nav.html [hunk 8]
```diff
@@ -3,23 +3,29 @@
...
-        {% for page in pages %}
+	   {% assign sorted-items = site.data.languages | sort: "name" %}
+       {% for item in sorted-items %}
           <div class="dropdown-item btn-icon">
             <div class="float-right">
-              <a href="{{ page.url }}" hreflang="{{ page.lang }}">{{ page.language }} {{ page.flag }}</a>
+               {% if item.tag == "en" %}
+               <a href="/">{{ item.name }} {{ item.flag }}</a>
+			   {% else %}
+			   <!-- we use the 'tag' variable in languages.yml to redirect to the correct page -->
```
Comment (wiz) [inline _includes/main_nav.html]: please remove this comment, it's getting output in the source code once per locale

File: _includes/main_nav.html [hunk 9]
```diff
@@ -28,18 +34,37 @@
...
       <ul class="navbar-nav ml-auto">
-
-          {% for item in site.data.main_nav.links %}
-          <li class="nav-item {% if page.url == item.url | append: '/' %}current{% endif %}">
+ 
+          {% for item in site.data.main_nav[site.lang].links %}
+          {% if site.lang == "en" %}
+            <li class="nav-item {% if page.url == item.url | append: '/' %}current{% endif %}">
+		  {% else %}
+		    <!-- concatenating site.baseurl (mulilangual plugin variable) with page.url to match item.url and underline green current page -->
```
Comment (wiz) [inline _includes/main_nav.html]: please remove this comment, it gets output into the page once per locale

File: _layouts/default.html [hunk 10]
```diff
@@ -29,15 +29,15 @@
         <meta property="twitter:site" content="{{ langdata.enabled }}">
         <meta name="twitter:creator" content="@bisq_network">
         <link rel="canonical" href="{{ page.url }}">
```
Comment (wiz) [inline _layouts/default.html:31]: canonical href should be the localized URL, for example /ja/dao/ when viewing /ja/dao/ - currently it's /dao/

File: _includes/main_nav.html [hunk 11]
```diff
@@ -3,23 +3,29 @@
     <a class="navbar-brand" href="/"><img src="/images/bisq-logo.svg" height="30"/></a>
 
     <!-- Language selector for mobile -->
-    {% if page.lang %}
     <div class="dropdown mobile-language-selector more-info-desktop">
       <button class="btn btn-sm dropdown-toggle" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-        {{ page.language }} {{ page.flag }}
+        {% for item in site.data.languages %}
+          {% if  site.lang == item.tag %} {{ item.name }} {{ item.flag }} {% endif %}
+        {% endfor %}
```
Comment (wiz) [inline _includes/main_nav.html]: this loop outputs about 10 newline characters, can you prevent that, maybe by joining these 3 lines into a single line?

File: _includes/main_nav.html [hunk 12]
```diff
@@ -49,31 +74,32 @@
...
                 <div class="float-right">
-                  <a href="/blog/">Blog</a>
+                  <a href="{{ '/blog/' | prepend: site.baseurl_root }}">{% t main_nav.liBlog %}</a>
                 </div>
               </div>
 
               <div class="dropdown-item btn-icon">
                 <div class="float-right">
-                  <a href="/community/">Community</a>
+                  {% if site.lang == "en" %}<a href="/community/">{% else %}<a href="/{{ site.lang }}/community/">{% endif %} {% t main_nav.liCommunity %}</a>
```
Comment (wiz) [inline _includes/main_nav.html]: Don't add an extra space before the `{% t main_nav.liCommunity %}` tag

**Comment by wiz** [comment]
Screen Shot 2019-10-08 at 21 32 38

**Comment by wiz** [comment]
Can you also "view source" of every page and fix the extra spaces, extra newlines, and inconsistent indentations as much as reasonably possible? For example, there are a lot of mixed hard-tabs and soft-tabs, and minor quirks in the indentations - we should try to keep it the same as the current live site or as uniform as possible. Other than those minor fixes, looks good 😀

**Comment by FKrauss** [comment]
but we can do that in a different merge, @wiz . I can do that after pushing the pt-br translation of the site

**Comment by wiz** [comment]
One more thing, seems you accidentally replaced "align-items-center" class with "align-t homepage_contents-center" class, you really gotta be careful when doing blanket search and replace like this 😓

File: index.html [hunk 13]
```diff
@@ -1,47 +1,42 @@
...
-ref: index
-lang: en
-flag: 🌐
-language: English
 ---
-{% assign item = site.data.en.homepage_content %}
 <div class="container fadeIn">
         <main class="bd-masthead" id="content" role="main">
-            <div class="row align-items-center">
+            <div class="row align-t homepage_contents-center">
```
Comment (wiz) [inline index.html]: accidental search/replace changed this CSS class

File: index.html [hunk 14]
```diff
@@ -170,49 +167,49 @@ <h1 class="intro text-monospace display-1" style="margin-left: -.3rem;">{{ item.
 
     <div class="row-block row">
 
-      <h2>{{ item.hFeatures }}</h2>
-      <div class="row align-items-center text-md-center text-left">
+      <h2>{% t homepage_content.hFeatures %}</h2>
+      <div class="row align-t homepage_contents-center text-md-center text-left">
```
Comment (wiz) [inline index.html]: accidental search/replace changed this CSS class

**Comment by erciccione** [comment]
@wiz thank you for your review, i'm implementing some changes now. I see a good amount of your comments are not related to the source code, but to the resulting jekyll built. Maybe it's my bad habit but i never really care too much about it, since it's mostly useful only for the reviewer (makes easier to compare the two builds, but nothing more) and for extremely minor things, like the duplicate comments.

> Can you also "view source" of every page and fix the extra spaces...

You are getting very nitpicky today :P. Please consider that with a major overhaul like the one introduced by this PR, it is normal that the resulting jekyll build (i guess you are referring to that since i don't see the problem in the source code) is slightly different from the precedent build. The point of this PR is to introduce a scalable system to develop on and easy to maintain. We are removing more than 60 duplicated files and editing 124. I think we can expect some minor (and irrelevant) differences.

> you really gotta be careful when doing blanket search and replace like this 

I used `sed` at the beginning to replace some strings, but i stopped immediatly after i saw there were too many variables to keep track of, this is a leftover from my cleanup. I apologize for forgetting it.

**Comment by erciccione** [comment]
I pushed some fixes. The proposed changes are all very minor, if @m52go agrees let's just get this merged and fix the missing issues in other PRs. I really don't see any reason to hold this work any longer.

**Comment by wiz** [comment]
The navbar-brand bug still reproduces with your latest pushed code deployed to the preview URL. To be clear, this is the issue I'm referring to:

[Steps to Reproduce]
1) Go to https://deploy-preview-260--bisq-website.netlify.com/ja/dao/
2) Click on Bisq logo in upper-left corner

[Expected Result]
Site behaves the same as live production site
I navigate to /ja/

[Actual Result]
Site behavior is different from live production site
I navigate to /

**Comment by wiz** [comment]
I found some more bugs with the mobile layout. What's going on with this navigation? It seems some extra links are using an older version of the Japanese translations, very weird 
Screen Shot 2019-10-08 at 23 56 21

**Comment by wiz** [comment]
Some html code got leaked into this string at the bottom of the community page
Screen Shot 2019-10-09 at 0 04 02

**Comment by erciccione** [comment]
The problem with tne mobile menu was caused by pre-existent conflicts which were revealed during one of the last rebases. Basically: strings for the nav menu are both in `main_nav.yml` and in the LANG.yml files. One of the two will have to be deleted, so that only one is used. In the meantime, i updated the japanese strings and using them from LANG.yml.

The HTML getting in seems to be a problem related only to the mobile version of the Japanese page. I will further investigate to see what it's causing it.

Fixed the issue with the logo in the top left.

**Comment by wiz** [comment]
@erciccione Hey can you test this yourself and fix all the remaining issues before @m52go and I re-review your PR again? For example, do this to diff the source code against the current live site:
```
curl https://deploy-preview-260--bisq-website.netlify.com/ja/dao/ > 260
curl https://bisq.network/ja/dao/ > live
diff -u live 260
```
I just did this again and there are still tons of issues, like this "/ja/" reference becoming "/ja":
```
-    <a class="navbar-brand" href="/ja/"><img src="https://d33wubrfki0l68.cloudfront.net/aecfa9ee0e182f2b64970b
1000095ae27311f991/61d30/images/bisq-logo.svg" height="30"/></a>
+   <a class="navbar-brand" href="/ja"><img src="https://d33wubrfki0l68.cloudfront.net/aecfa9ee0e182f2b64970b10
00095ae27311f991/61d30/images/bisq-logo.svg" height="30"/></a>
```
Or the hreflang parameter disappearing from this <a> tag:
```
-              <a href="/dao/" hreflang="en">English 🌐</a>
+               
+               <a href="/">English 🌐</a>
+			   
```
And maybe 5 or 10 more like this. You will also see why the whitespace and indentation are not "nitpicking" - it makes the diff very difficult to check. Please make the output from this PR as identical as possible to the live site, it shouldn't really change anything at all, there shouldn't even be a diff to check :)

**Comment by erciccione** [comment]
I think i'm a bit confused. What is the problem we are trying to solve? Could you point out a problem or a difference in the final output caused by those "issues"?

Also, what do you think about fixing minor or irrelevant differences, like the different hreflang, in a subsequent PR, or are these issues important enough to keep this one on hold? If it's the latter, i try to work on it tomorrow but i'm not sure of how much time i will have.

**Comment by m52go** [comment]
@erciccione I think the point wiz is trying to make is that if this is a refactor, the resulting built code base and functionality should be the same. In some cases it matters more than others...but from the not-irrelevant errors wiz has uncovered (e.g., stray HTML showing on page, mobile menu items differences, old translations, etc), I start to wonder about the integrity of the proposed changes (i.e., what else is there that we are not catching?).

Minor or major, why is it that we're seeing such errors if this PR is supposed to essentially be a refactor of how/which strings are delivered to the user? Not at all saying any of this is trivial, but I'm not understanding how the resulting differences in this PR came about...and they do make me a bit uneasy.

For what it's worth, I will be devising a process to more rigorously review PRs from now on.

**Comment by wiz** [comment]
> What is the problem we are trying to solve?

While reviewing this PR, we realized there might be flaws in the PR review process since so many issues were missed.

For starters, judging from the amount of changes requested so far that seem to be obvious bugs, it seems this PR wasn't tested before being submitted for review, or each time before re-review. Also, after reviewers pointed out the seemingly unintentional changes introduced by this PR, it's not clear to everybody why unintentionally changing things is bad, so now I think the problem at hand is to reach consensus on how exactly to review this PR and review/approve these changes one by one.

To solve this, I'm proposing for us to use a very pragmatic before/after diff of the website output, in addition to visually reviewing the PR output, and review those changes one by one, so we can see exactly what might be changing "under the hood" in addition to the obvious visual changes. In addition, I think it would be a good idea if PR submitters do this themselves before submitting the PR, so they can identify and fix unintentional changes before the review process.

> Could you point out a problem or a difference in the final output caused by those "issues"?

Sure, but I think I've already pointed out enough of your issues for you, so for now I'm asking you to *test it yourself* and check the diff output before we re-review. Ideally a refactoring should be exactly the same in terms of functionality and output changes, the goal should be to have a null diff. If there are exceptions to that they need to be reviewed and approved one by one.

But if it wasn't clear enough, I'll take a screenshot of my previous comment and highlight all the changes in red and try to clarify the differences in this one example as much as possible:

Screen Shot 2019-10-09 at 14 53 09

1) The `href="/dao/"` has changed to `href="/"`

This is a different URL, which means the user will go to a different page. If you already fixed this one, great :+1:

2) The `hreflang="en"` has been removed

I added this tag to improve SEO a couple of months ago, and it fixed the issues we were having on Google. It's not cool if you just randomly remove it as part of this PR. https://github.com/bisq-network/bisq-website/pull/215/files#diff-8a279082e95baa3b49f6a99e29febe18R16 

3) The indentation has been changed, and additional newlines have been added

While indentation and newlines may not seem important to you, they are very important when checking what has actually changed. If you arbitrarily change tabs into spaces, this is very bad for version control and reviewing changes.

I could go on, but since there are 10+ of these diff snippets that need to be reviewed individually, and since the reviewers of this PR have already spent considerable time, I'm now asking you to go back and test and improve the PR further on your own using the diff output process, in order to reduce the amount of unintentional changes, before we re-review it again. I think this would have saved the reviewers a lot of time pointing out the obvious UI issues to you - this thread is 79 comments so far, so I think we've spent enough time reviewing for now.

> Also, what do you think about fixing minor or irrelevant differences, like the different hreflang, in a subsequent PR, or are these issues important enough to keep this one on hold?

Well, that's exactly my point, isn't it? All of "these issues" need to be reviewed, one by one, and only after we identify and discuss each of them, can we decide if they are important or not. Currently, since nobody even knew about them until I pointed this out, it means we weren't properly reviewing the PR and need to change our review process.

**Comment by erciccione** [comment]
Some parts are looking different beacause they are different. This is not just a refactor, i had to edit files and change things, not just add variables and functions. As i already explained in past (for example in https://github.com/bisq-network/bisq-website/pull/248#issuecomment-533110278) the situation of this repository is really bad and apparently it only got patches on the top of a broken system, so excuse me if i'm a bit weirded out when i see as a pr-stopper "there are a lot of newlines" (not even in the source code, in that case this complaint would make sense, but in the resulting built code).

@wiz veiled provocations are starting to be very annoying. I never intended to leave something out, just to include it in a different PR to not clutter this one with minor changes (even if i understand that when the repo it's directly deployed to master, you want to have the PR as perfect as possible. This is IMO an issue: see #261 ). Also, I did heavily tested this work and i see that i made some gross mistakes. I apologize for those, i don't have much experience as a web developer and this work was a useful lesson that allowed me to find some problems in my workflow (which are partly responsible for some mistakes).

That doesn't change the fact that the way this PR is being reviewed is a nightmare.
Seems like most of the submitted reviews are based on the built website, not on the actual source code and they are mostly posted using *screenshots*. Even after i asked to stop because they were making working on this much harder. In some cases even screenshots of the built code! which i had to figure out were it came from, wasting a considerable amount of time. There are even screenshots of text in Japanese, which i had to figure out where the hell was coming from, because i cannot just grep text from a screenshot and i definitely cannot write japanese characters.
I accepted this inefficient and flawed workflow just for the sake of finally having a scalable and structured website as a good base to work on. I agree with you @m52go, this repo definitely need a better reviewing process.

For example: 

> there are still tons of issues, like this "/ja/" reference becoming "/ja"

My question that is still unanswered is: why is this an issue? I would totally understand if it's breaking something (that link seems to work properly), like other issues that were correctly rised, but looks like the only reason to point that out is that the reference is different from the precedent built source code. Maybe something changed in the way that reference is dealt with, and the final slash is not necessary anymore? It's extremely weird that i'm asked to change it back just because it's different, without telling me what it's even breaking. Please tell me if i'm missing something.

If my work has problems and you intend to review in a reasonable way, i'm happy to fix the issues when pointed out (like the problems of the missing hreflang and the html leaking in that japanese mobile page), otherwise - and if the "integrity" of this work doesn't appeal you or you consider it of low quality - please let me know and i will just close this PR.

**Pull request closed without merging**